src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorStatModule.java

```diff
@@ -0,0 +1,168 @@
+    BooleanLoadable.INSTANCE.defaultField("allow_broken", false, MaxArmorStatModule::allowBroken),
+    Loadables.ITEM_TAG.nullableField("held_tag", MaxArmorStatModule::heldTag),
+    new EnumLoadable<>(ArmorStatModule.TooltipStyle.class).defaultField("tooltip_style", ArmorStatModule.TooltipStyle.NONE, MaxArmorStatModule::tooltipStyle),
+    ModifierCondition.TOOL_FIELD,
+    MaxArmorStatModule::new);
+
+  @Override
+  public RecordLoadable<MaxArmorStatModule> getLoader() {
+    return LOADER;
+  }
+
+  @Override
+  public void addModules(ModuleHookMap.Builder builder) {
+    builder.addModule(new MaxArmorLevelModule(key2, allowBroken, heldTag));
+  }
+
+  @Override
+  public List<ModuleHook<?>> getDefaultHooks() {
+    return tooltipStyle == ArmorStatModule.TooltipStyle.NONE ? NO_TOOLTIP_HOOKS : TOOLTIP_HOOKS;
+  }
+
+  @Override
+  public void onEquip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
+    if (condition.matches(tool, modifier)) {
+      context.getTinkerData().ifPresent(data -> {
+        addStatIfArmor(tool, context, key, amount.compute(data.get(key2, 0)), allowBroken, heldTag);
+      });
+
+    }
+  }
+
+  @Override
+  public void onUnequip(IToolStackView tool, ModifierEntry modifier, EquipmentChangeContext context) {
+    if (condition.matches(tool, modifier)) {
+      context.getTinkerData().ifPresent(data -> {
+        addStatIfArmor(tool, context, key, amount.compute(data.get(key2, 0)), allowBroken, heldTag);
+      });    }
+  }
+
+  @Override
+  public void addTooltip(IToolStackView tool, ModifierEntry modifier, @Nullable Player player, List<Component> tooltip, TooltipKey tooltipKey, TooltipFlag tooltipFlag) {
+    if (condition.matches(tool, modifier) && (tool.hasTag(TinkerTags.Items.WORN_ARMOR) || heldTag != null && tool.hasTag(heldTag)) && (!tool.isBroken() || allowBroken)) {
+      float value = amount.compute(modifier.getEffectiveLevel());
+      if (value != 0) {
+        Component name = Component.translatable(Util.makeTranslationKey("armor_stat", key.getId()));
+        switch (tooltipStyle) {
+          case BOOST -> TooltipModifierHook.addFlatBoost(modifier.getModifier(), name, value, tooltip);
+          case PERCENT -> TooltipModifierHook.addPercentBoost(modifier.getModifier(), name, value, tooltip);
+        }
+      }
+    }
+  }
+
+  /**
+   * Adds to the armor stat for the given key. Make sure to subtract on unequip if you add on equip, it will not automatically be removed.
+   * @param context  Equipment change context
+   * @param key      Key to modify
+   * @param amount   Amount to add
+   */
+  public static void addStat(EquipmentChangeContext context, TinkerDataCapability.TinkerDataKey<Float> key, float amount) {
+    context.getTinkerData().ifPresent(data -> {
+      float totalLevels = data.get(key, 0f) + amount;
+      if (totalLevels <= 0.005f) {
+        data.remove(key);
+      } else {
+        data.put(key, totalLevels);
+      }
+    });
+  }
```

Comment:
This is wrong. This will entirely overwrite every other usage of that stat. What you want to do is have an int key which keeps track of the level, and then update the float based on the change in the int key. Call the helpers in `MaxArmorLevelModule` to make that work